sw/host/hsmtool/src/util/wrap.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+// This module provides functionality for wrapping and unwrapping cryptographic
+// keys using different mechanisms. It supports AES key wrap, AES key wrap with
+// padding, and RSA PKCS OAEP.
+
+use anyhow::{Ok, Result};
+use cryptoki::mechanism::rsa::{PkcsMgfType, PkcsOaepParams, PkcsOaepSource};
+use cryptoki::mechanism::{Mechanism, MechanismType};
+use cryptoki::object::{Attribute, ObjectHandle};
+use cryptoki::session::Session;
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+
+use crate::error::HsmError;
+use crate::util::attribute::{AttributeMap, AttributeType, KeyType, ObjectClass};
+use crate::util::helper;
+
+/// The wrapping mechanism to use.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+pub enum Wrap {
+    #[serde(alias = "aes-key-wrap")]
+    AesKeyWrap,
+    #[serde(alias = "aes-key-wrap-pad")]
+    AesKeyWrapPad,
+    #[serde(alias = "rsa-pkcs-oaep")]
+    RsaPkcsOaep,
+}
+
+impl FromStr for Wrap {
+    type Err = anyhow::Error;
+    fn from_str(input: &str) -> Result<Self> {
+        if input.eq_ignore_ascii_case("aes-key-wrap") {
+            Ok(Wrap::AesKeyWrap)
+        } else if input.eq_ignore_ascii_case("aes-key-wrap-pad") {
+            Ok(Wrap::AesKeyWrapPad)
+        } else if input.eq_ignore_ascii_case("rsa-pkcs-oaep") {
+            Ok(Wrap::RsaPkcsOaep)
+        } else {
+            Err(HsmError::Unsupported(format!("invalid variant: {input}")).into())
+        }
+    }
+}
```

Comment:
FYI, for simple enums (e.g. enums that don't have embedded structures), the `strum` crate can derive `FromStr` for you.

If you're going to use this as a CLI argument, you might look at `clap::ValueEnum`.  ValueEnum also supplies the list of variants to the clap `Args` derive so you don't have manually specify the values in a `HELP` string.

The reason the `Format` enum doesn't use either of these is because it is a complex enum that can carry parameters (e.g. the Slice variant carries a range).

Reply:
Thanks for the feedback! This simplified the code.